packages/react-dom/src/__tests__/ReactDOMHydrationDiff-test.js

```diff
@@ -330,6 +390,14 @@ describe('ReactDOMServerHydration', () => {
         It can also happen if the client has a browser extension installed which messes with the HTML before React loaded.
 
         https://react.dev/link/hydration-mismatch
+
+          <Mismatch isClient={true}>
+            <div className="parent">
+              <main
+                className="child"
+        +       style={{opacity:1}}
+        -       style={{opacity:"0"}}
```

Comment:
Does this imply to users that part of the problem is one is a string and the other is a number? I wonder if we should try to parse the html attribute as a number if the hydration prop is also a number. It'd have to be clever like 

```
if (typeof prop === 'number' && parseInt(attr).toString() === attr) {
  return parseInt(attr)
} else {
  return attr
}
```

Reply:
unrelated, but maybe worth adding a space for readability? `opacity: 1`

Reply:
Yes but I think you'll find that this is not the only place we could be clever. The more important one might be to find equivalences of values. It'll also ignore any style prop that isn't parsed since we read back the styles using the browser. So a browser specific one might show up as diff if something else in the stylesheet differences.

I wouldn't over rotate on the unit tests since they're just the ones that used to be here and all the other ones don't use snapshots that test this. There's more interesting cases to be found in production code.

I ended up not including spaces for these cases because the goal is to keep it compact enough to fit on one line. Especially in the case where these are formatted on contextual nodes because I always enforce it to take up at most one line so a few extra spaces can mean that a prop gets dropped instead. Don't really want to do a constraint solving where it tries to fit it.